### PR TITLE
add type hint to seed argument in torch.random.manual_seed

### PR DESCRIPTION
**Pull request by orkhank**
The type hint for the `seed` argument is not specified in the function signature, which makes the function "partially unknown".

Especially in strict mode, type checkers do not like that:
```python
from typing import reveal_type

import torch.random
torch.random.manual_seed(42)   # Pylance(reportUnknownMemberType): Type of "manual_seed" is partially unknown

reveal_type(torch.random.manual_seed)  # Type of "torch.random.manual_seed" is "(seed: Unknown) -> Generator"
```

Since the type of `seed` is specified as `int` in the _docstring_, the function signature is updated accordingly.

All my homies hate the random squiggly lines in their IDEs. This PR goes out to them.

### CODE REVIEW

File: torch/random.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def get_rng_state() -> torch.Tensor:
     return default_generator.get_state()
 
 
-def manual_seed(seed) -> torch._C.Generator:
+def manual_seed(seed: int) -> torch._C.Generator:
```
Comment (Skylion007) [inline torch/random.py:49]: Technically, the typing here is IntLike protocol isn't it? Do we have a typing Protocol for this already I wonder.

Comment (orkhank) [reply torch/random.py:49]: I am not sure...

I tried to look for such a type, but could not find a suitable one.[^1]

[^1]: Although, there is [`IntLikeType`](https://github.com/pytorch/pytorch/blob/a878d84f364e459a5538f8c794f85cf6b7de0fb7/torch/types.py#L55).
It is often used in the `distributed` module (exhibit [A](https://github.com/pytorch/pytorch/blob/a878d84f364e459a5538f8c794f85cf6b7de0fb7/torch/distributed/tensor/_random.py#L393), [B](https://github.com/pytorch/pytorch/blob/a878d84f364e459a5538f8c794f85cf6b7de0fb7/torch/distributed/tensor/_collective_utils.py#L216), [C](https://github.com/pytorch/pytorch/blob/a878d84f364e459a5538f8c794f85cf6b7de0fb7/torch/distributed/tensor/_redistribute.py#L1200)).
So I am not sure, if they are intended to be used for the humble _seed_.

Then, I went over >1100 calls to `manual_seed` in the repo.
Almost all use either integer literals (>970 instances) or some `int | Unknown` type variable.

I think there are just three places where a non-`int` values is passed.
Those use the integer types from numpy and cause the CI lint check to fail.
Besides the three problematic cases, I did not encounter any other examples that imply `manual_seed` should have a more permissive argument type.

For the sake of simplicity, I think we can just safely convert the numpy values to `builtin.int` with `int(...)`.
Here, is a similar pattern from the repo for [`Generator.manual_seed`](https://docs.pytorch.org/docs/2.12/generated/torch.Generator.html#torch.Generator.manual_seed):
https://github.com/pytorch/pytorch/blob/a878d84f364e459a5538f8c794f85cf6b7de0fb7/torch/_functorch/_aot_autograd/runtime_wrappers.py#L2445-L2453

With commit e9080ddd0a6f179677d7c68f381ffe67b29bd8f3, I fix the CI lint issues using the approach above;
would appreciate your review.

Still, if these changes are not okay, can you please direct me towards a more concrete type for the _IntLike protocol_?
